Testnet/DAOContract/DAOContract/DAOContract.cs

```diff
@@ -0,0 +1,213 @@
+﻿using Stratis.SmartContracts;
+using System;
+
+[Deploy]
+public class DAOContract : SmartContract
+{
+    public Address Owner
+    {
+        get => State.GetAddress(nameof(Owner));
+        private set => State.SetAddress(nameof(Owner), value);
+    }
+    /// <summary>
+    /// Min quorum for yes votes
+    /// </summary>
+    public uint MinQuorum
+    {
+        get => State.GetUInt32(nameof(MinQuorum));
+        private set => State.SetUInt32(nameof(MinQuorum), value);
+    }
+
+    public uint LastProposalId
+    {
+        get => State.GetUInt32(nameof(LastProposalId));
+        private set => State.SetUInt32(nameof(LastProposalId), value);
+    }
+
+    public bool IsWhitelisted(Address address) => State.GetBool($"Whitelisted:{address}");
+    private void SetIsWhitelisted(Address address, bool allowed) => State.SetBool($"Whitelisted:{address}", allowed);
+
+    public ulong GetVotingDeadline(uint proposalId) => State.GetUInt64($"Deadline:{proposalId}");
+    private void SetVotingDeadline(uint proposalId, ulong block) => State.SetUInt64($"Deadline:{proposalId}", block);
+
+    public uint GetYesVotes(uint proposalId) => State.GetUInt32($"YesVotes:{proposalId}");
+    private void SetYesVotes(uint proposalId, uint value) => State.SetUInt32($"YesVotes:{proposalId}", value);
+
+    public uint GetNoVotes(uint proposalId) => State.GetUInt32($"NoVotes:{proposalId}");
+    private void SetNoVotes(uint proposalId, uint value) => State.SetUInt32($"NoVotes:{proposalId}", value);
+
+    public uint GetVote(uint proposalId, Address address) => State.GetUInt32($"Vote:{proposalId}:{address}");
+
+    public Proposal GetProposal(uint index) => State.GetStruct<Proposal>($"Proposals:{index}");
+
+    private void SetProposal(uint index, Proposal proposal) => State.SetStruct($"Proposals:{index}", proposal);
+
+    public DAOContract(ISmartContractState state, uint minQuorum)
+        : base(state)
+    {
+        this.MinQuorum = minQuorum;
+        this.Owner = Message.Sender;
+        this.LastProposalId = 1;
+    }
+
+    public uint CreateProposal(Address recipent, ulong amount, uint votingDuration, string description)
+    {
+        var proposal = new Proposal
+        {
+            RequestedAmount = amount,
+            Description = description,
+            Recipient = recipent,
+            Owner = Message.Sender
+        };
+
+        SetProposal(LastProposalId, proposal);
+        SetVotingDeadline(LastProposalId, checked(votingDuration + Block.Number));
+        Log(new ProposalAddedLog
+        {
+            ProposalId = LastProposalId,
+            Recipent = recipent,
+            Amount = amount,
+            Description = description
+        });
+
+        return LastProposalId++;
```

Comment:
If you want to save a small amount of gas here, set `var lastProposalId = LastProposalId;` near the top and reuse. Then you can use the below to return out. 
```
LastProposalId = lastProposalId++;
return lastProposalId;
```